2009/serial/E/tool.cpp

```diff
@@ -0,0 +1,169 @@
+                             it4 != (it1->first).end(); it4++)
+                        {
+                            s.insert(*it4);
+                            k.insert(*it4);
+                        }
+                        s.insert(*it2);
+                        k.insert(*it2);
+                        k.insert(*it3);
+                        int sub_ind = 1;
+                        sub_ind = is_ind(k, superset);
+                        if (sub_ind)
+                            next[s].insert(*it3);
+                        s.clear();
+                    }
+                }
+            }
+        }
+        if (next.size() == 0)
+        {
+            for (map<set<unsigned int>, set<unsigned int>>::iterator it1 =
+                     ind.begin();
+                 it1 != ind.end(); it1++)
+            {
+                for (set<unsigned int>::iterator it2 = (it1->second).begin();
+                     it2 != (it1->second).end(); it2++)
+                {
+                    set<unsigned int> s;
+                    for (set<unsigned int>::iterator it3 = (it1->first).begin();
+                         it3 != (it1->first).end(); it3++)
+                        s.insert(*it3);
+                    s.insert(*it2);
+                    output.insert(s);
+                }
+            }
+        }
+        ind.clear();
+        for (map<set<unsigned int>, set<unsigned int>>::iterator it1 =
+                 next.begin();
+             it1 != next.end(); it1++)
+        {
+            for (set<unsigned int>::iterator it2 = (it1->second).begin();
+                 it2 != (it1->second).end(); it2++)
+            {
+                set<unsigned int> s;
+                for (set<unsigned int>::iterator it3 = (it1->first).begin();
+                     it3 != (it1->first).end(); it3++)
+                    s.insert(*it3);
+                ind[s].insert(*it2);
+                s.clear();
+            }
+        }
+        next.clear();
+    } while (ind.size() > 0);
+}
+
+void count(const set<set<unsigned int>>& superset, const unsigned int n)
+{
+    map<set<unsigned int>, set<unsigned int>> ind;
+    to_ind(n, superset, ind);
+    search(superset, ind);
+}
+
+int main(int argc, char* argv[], char* envp[])
+{
+    struct timeval timevalA;
+    struct timeval timevalB;
+    unsigned int n;
+    set<set<unsigned int>> superset;
+
+    FILE* f = stdin;
+    fscanf(f, "%d", &n);
+    while (!feof(f))
+    {
+        unsigned int a, b;
+        fscanf(f, "%d", &a);
+        fscanf(f, "%d", &b);
+        set<unsigned int> s;
+        s.insert(a);
+        s.insert(b);
+        superset.insert(s);
+        s.clear();
+    }
+    fclose(f);
+
+    gettimeofday(&timevalA, NULL);
+    count(superset, n);
+    gettimeofday(&timevalB, NULL);
+
+    cout << output.size() << endl;
+
+    cout << "time="
+         << timevalB.tv_sec - timevalA.tv_sec +
+                (timevalB.tv_usec - timevalA.tv_usec) / (double)1000000
+         << endl;
+
```

Comment:
I got the following error, while trying to compile it: 
`tool.cpp:11:21: error: a space is required between consecutive right angle brackets (use '> >')
tool.cpp:13:55: error: a space is required between consecutive right angle brackets (use '> >')
tool.cpp:14:52: error: a space is required between consecutive right angle brackets (use '> >')
tool.cpp:32:65: error: a space is required between consecutive right angle brackets (use '> >')
tool.cpp:52:39: error: a space is required between consecutive right angle brackets (use '> >')
tool.cpp:53:52: error: a space is required between consecutive right angle brackets (use '> >')
tool.cpp:57:48: error: a space is required between consecutive right angle brackets (use '> >')
tool.cpp:58:53: error: a space is required between consecutive right angle brackets (use '> >')
tool.cpp:92:57: error: a space is required between consecutive right angle brackets (use '> >')
tool.cpp:109:53: error: a space is required between consecutive right angle brackets (use '> >')
tool.cpp:128:38: error: a space is required between consecutive right angle brackets (use '> >')
tool.cpp:130:44: error: a space is required between consecutive right angle brackets (use '> >')
tool.cpp:140:25: error: a space is required between consecutive right angle brackets (use '> >')`

You can fix it by replacing all occurrences of  '>>' by '> >'.

Reply:
Please, try to compile them again